ethereum/spec/src/main/java/tech/pegasys/teku/spec/logic/versions/altair/forktransition/AltairStateUpgrade.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.spec.logic.versions.altair.forktransition;
+
+import tech.pegasys.teku.infrastructure.unsigned.UInt64;
+import tech.pegasys.teku.spec.config.SpecConfigAltair;
+import tech.pegasys.teku.spec.datastructures.state.Fork;
+import tech.pegasys.teku.spec.datastructures.state.SyncCommittee;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.BeaconState;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.common.BeaconStateFields;
+import tech.pegasys.teku.spec.datastructures.state.beaconstate.versions.altair.BeaconStateAltair;
+import tech.pegasys.teku.spec.logic.common.forktransition.StateUpgrade;
+import tech.pegasys.teku.spec.logic.versions.altair.helpers.BeaconStateAccessorsAltair;
+import tech.pegasys.teku.spec.schemas.SchemaDefinitionsAltair;
+import tech.pegasys.teku.ssz.primitive.SszByte;
+import tech.pegasys.teku.ssz.primitive.SszUInt64;
+
+public class AltairStateUpgrade implements StateUpgrade<BeaconStateAltair> {
+  final SpecConfigAltair specConfig;
+  final SchemaDefinitionsAltair schemaDefinitions;
+  final BeaconStateAccessorsAltair beaconStateAccessors;
+
+  public AltairStateUpgrade(
+      final SpecConfigAltair specConfig,
+      final SchemaDefinitionsAltair schemaDefinitions,
+      final BeaconStateAccessorsAltair beaconStateAccessors) {
+    this.specConfig = specConfig;
+    this.schemaDefinitions = schemaDefinitions;
+    this.beaconStateAccessors = beaconStateAccessors;
+  }
+
+  @Override
+  public BeaconStateAltair upgrade(final BeaconState preState) {
+    final UInt64 epoch = beaconStateAccessors.getCurrentEpoch(preState);
```

Comment:
When we integrate this we need to be sure that the `preState` has already processed the epoch transition into Altair.  I'm not sure if that transition should be done on the old or new specs though.  Worst case we'll need to change this to have a `.plus(1)` if the preState is actually from the epoch before Altair starts.

Reference tests will easily pick up that kind of problem thankfully.